openwisp_controller/config/tests/test_api.py

```diff
@@ -389,6 +427,17 @@ def test_vpn_list_api(self):
             r = self.client.get(path)
         self.assertEqual(r.status_code, 200)
 
+    def test_vpn_list_for_shared_objects(self):
+        self._create_ca(name='shared_ca', organization=None)
+        self._create_cert(name='shared_cert', organization=None)
+        org1 = self._get_org()
+        test_user = self._create_operator(organizations=[org1])
+        self.client.force_login(test_user)
+        path = reverse('controller_config:api_vpn_list')
+        r = self.client.get(path, {'format': 'api'})
+        self.assertContains(r, 'shared_ca</option>')
+        self.assertContains(r, 'shared_cert</option>')
+
```

Comment:
I don't see a test which ensures shared templates can be assigned to a device, was that already added before? If not, can you please add it?

Reply:
Yes, It is contained in this:- https://github.com/ManishShah120/openwisp-controller/blob/4eacd3cca7e06e8a9f84153ef966bb0601d765c1/openwisp_controller/config/tests/test_api.py#L203-L215
```python
def test_device_patch_with_templates_of_same_org(self):
    org1 = self._create_org(name='testorg')
    d1 = self._create_device(name='org1-config', organization=org1)
    self._create_config(device=d1)
    self.assertEqual(d1.config.templates.count(), 0)
    path = reverse('controller_config:api_device_detail', args=[d1.pk])
    t1 = self._create_template(name='t1', organization=None)
    t2 = self._create_template(name='t2', organization=org1)
    data = {'config': {'templates': [str(t1.id), str(t2.id)]}}
    r = self.client.patch(path, data, content_type='application/json')
    self.assertEqual(r.status_code, 200)
    self.assertEqual(d1.config.templates.count(), 2)
    self.assertEqual(r.data['config']['templates'], [t1.id, t2.id])
```